AR1/reduction/interferometric_pointing/analyse_interferometric_pointing.py

```diff
@@ -244,6 +232,33 @@ def _eq_TGT_(tgt): # tgt==TGT, "tags" don't matter
             g.close()
         return ant_pointing
 
+def load_rfi_static_mask(filename, freqs, debug=True, chunks=1):
+    # Construct a mask either from a pickle file, or a text file with frequency ranges
+    nchans = len(freqs)
+    try:
+        with open(filename, "rb") as pickle_file:
+            channel_flags = pickle.load(pickle_file)
+        nflags = len(channel_flags)
+        if (nchans != nflags):
+            print("Warning channel mask (%d) is stretched to fit dataset (%d)!"%(nflags,nchans))
+            N = nchans/float(nflags)
+            channel_flags = np.repeat(channel_flags, int(N+0.5)) if (N > 1) else channel_flags[::int(1/N)]
+        channel_flags = channel_flags[:nchans] # Clip, just in case
+    except pickle.UnpicklingError: # Not a pickle file, perhaps a plain text file with frequency ranges in MHz?
+        mask_ranges = np.loadtxt(filename, comments='#', delimiter=',')
+        channel_flags = np.full((nchans,), False)
+        for r in mask_ranges:
+            idx = np.where((freqs < r[1]*1e6) & (freqs >= r[0]*1e6))[0]
```

Comment:
There is still a small issue here to do with the channel width... In `katsdpmodels` we flag a channel if any part of it overlaps with a mask range, while you are checking the channel centres only. Our check looks something like this:
```Python
        low = freqs - 0.5 * channel_width
        high = freqs + 0.5 * channel_width
        in_range = (min_freq <= high) & (low <= max_freq)
```
But this is a minor issue. In the near future we could even leverage `katsdpmodels` directly here, once it is integrated with `katdal`.